dataflux_pytorch/dataflux_mapstyle_dataset.py

```diff
@@ -136,8 +138,11 @@ def __init__(
         self.bucket_name = bucket_name
         self.data_format_fn = data_format_fn
         self.config = config
-        # If composed download is enabled, check if the client has permissions to create and delete the composed object.
-        if self.config.max_composite_object_size != 0:
+        # If composed download is enabled an a storage_client was provided,
+        # check if the client has permissions to create and delete the
+        # composed object.
+        if storage_client is not None \
+                and self.config.max_composite_object_size != 0:
```

Comment:
to be pythonic, this line can just be `and self.config.max_compose_object_size` as zero will default to false and any non-zero value will default to true